File: test/fakers/brand_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/catalog/dtos/brand_dto.dart';

class BrandFaker {
  static final faker = Faker();

  static BrandDto fakeDto({String? id, String? name}) {
    return BrandDto(
      id: id ?? faker.guid.guid(),
      name: name ?? faker.company.name(),
    );
  }

  static List<BrandDto> fakeManyDto({
    int count = 10,
    String? id,
    String? name,
  }) {
    return List.generate(count, (index) => fakeDto(id: id, name: name));
  }
}

File: sertton-react-native-app/src/core/catalog/entities/fakers/brand-faker.ts
import { faker } from "@faker-js/faker"

import type { Brand } from "../brand"

export class BrandFaker {
  static fake(brand?: Partial<Brand>): Brand {
    return { id: faker.string.uuid(), name: faker.company.name(), ...brand }
  }

  static fakeMany(count = 10, brand?: Partial<Brand>): Brand[] {
    return Array.from({ length: count }, () => BrandFaker.fake(brand))
  }
}
